app/processors/processor_base.py

```diff
@@ -0,0 +1,74 @@
+from app import settings
+from app.helpers.request_helper import remote_call, response_ok, get_sequence_no
+from sdc.rabbit.exceptions import QuarantinableError, RetryableError
+
+
+class Processor:
+    """Base class calls transformer and passes results to ftp"""
+
+    def __init__(self, logger, survey, ftpconn, base_url, endpoint_name):
+        self.logger = logger
+        self.survey = survey
+        self.tx_id = ""
+        self._setup_logger()
+        self.ftp = ftpconn
+        self._base_url = base_url
+        self._endpoint_name = endpoint_name
+
+    def process(self):
+        """call transform and error if needed"""
+        transformed = self._transform()
+
+        delivered = self.ftp.unzip_and_deliver(self._get_ftp_folder(self.survey), transformed)
+
+        if not delivered:
+            self.logger.error("Failed to deliver zip to ftp")
+            raise RetryableError("Failed to deliver zip to ftp")
+        return
+
+    def _transform(self):
+        """ call the transform endpoint and raise quarantinable error if bad response"""
+        endpoint = self._get_url()
+        self.logger.info("Calling transform", request_url=endpoint)
+
+        response = remote_call(endpoint, json=self.survey)
+
+        if response_ok(response) and response.content is not None:
+            self.logger.info("{}:Successfully transformed".format(self.__class__.__name__))
+            return response.content
+        else:
+            raise QuarantinableError("Response missing content")
+
+    def _get_url(self):
+        """Gets the transformer url"""
```

Comment:
Wondering if this method would be better on the child classes and then you wouldn't need endpoint name at all as it would become part of the string, could also simplify the init method then too.

Reply:
I did have it like that for a while , but since there is no concept of abstract methods in python ( without using the ABC Metaclass) It meant that the ide marked the calls to _get_url as being not implemented as they were fixed up at run time .  Not a biggie , I could go either way .